src/Language/PureScript/TypeChecker.hs

```diff
@@ -124,17 +124,23 @@ addTypeClass
   -> m ()
 addTypeClass qualifiedClassName args implies dependencies ds = do
   env <- getEnv
-  traverse_ (checkMemberIsUsable (typeSynonyms env)) classMembers
+  let newClass = mkNewClass env
+  traverse_ (checkMemberIsUsable newClass (typeSynonyms env)) classMembers
   modify $ \st -> st { checkEnv = (checkEnv st) { typeClasses = M.insert qualifiedClassName newClass (typeClasses . checkEnv $ st) } }
   where
     classMembers :: [(Ident, Type)]
     classMembers = map toPair ds
 
-    newClass :: TypeClassData
-    newClass = makeTypeClassData args classMembers implies dependencies
+    mkNewClass :: Environment -> TypeClassData
+    mkNewClass env = makeTypeClassData args classMembers implies dependencies ctIsEmpty
+      where
+      ctIsEmpty = null classMembers && all (typeClassIsEmpty . findSuperClass) implies
```

Comment:
If there is a superclass, is it possible that we try to solve a constraint for the superclass when only an instance for the subclass is in scope, and then by looking up the superclass as a property on the subclass dictionary, we run into a problem at runtime?

Reply:
I guess you might need a chain of two or more superclasses for it to become an issue though, otherwise you'll just get `undefined` anyway.

Reply:
If there is a non-empty super class, then this class isn't considered empty.  So there shouldn't be any issues with looking up super classes, as I don't mark those ones as unused.

Reply:
I think it would be worth adding a test for this though (and maybe a few other cases) just to make sure we don't break that assumption in future. To be clear, the reasoning is that even if the subclass and superclass are empty, we'll never look up the superclass dictionary at runtime because it would be empty if we did. Right?

Reply:
You're right, I need to think more about subclass lookup.  I think the fix might just be moving the `Unused` hint slightly further up the dictionary expression.  Thanks!